store/heightsub.go

```diff
@@ -89,17 +91,13 @@ func (hs *heightSub[H]) Pub(headers ...H) {
 		return
 	}
 
-	height := hs.Height()
 	from, to := headers[0].Height(), headers[ln-1].Height()
-	if height+1 != from && height != 0 { // height != 0 is needed to enable init from any height and not only 1
```

Comment:
Took from https://github.com/celestiaorg/go-header/pull/197

Basically this was the last thing that was requiring adjacency in `store`.